File: PhoneDirectory.py
class PhoneDirectory:
    def __init__(self, maxNumbers):
        self.curr=0
        self.numbers=[i for i in range(maxNumbers)]
        self.used=[False]*maxNumbers
    def get(self):
        if self.curr==len(self.numbers):
            return -1
        number=self.numbers[self.curr]
        self.curr+=1
        self.used[number]=True
        return number
    def check(self, number):
        return 0<=number<len(self.numbers) and not self.used[number]
    def release(self, number):
        if(self.used[number]):
            self.curr-=1
            self.numbers[self.curr] = number
            self.used[number]=False
            
obj=PhoneDirectory(5)
print(len(obj.numbers))
print(obj.get())

print(obj.used)
print(obj.get())

print(obj.used)
print(obj.check(1))
obj.release(1)
print(obj.check(1))

File: PhoneDirectory.js
class PhoneDirectory{
    constructor(maxNumber){
        this.curr=0
        this.numbers=[]
        this.used=[]
        for(let i=0;i<maxNumber;i++){
            this.numbers.push(i)
            this.used.push(false)
        }
    }
    get(){
        if(this.curr==this.numbers.length){
            return -1
        }
        let number= this.numbers[this.curr]
        this.curr+=1
        this.used[number]=true
        return number
    }
    check(number){
        return (0<=number && number<this.numbers.length) && (!this.used[number])
    }
    release(number){
        if(this.used[number]){
            this.curr-=1
            this.numbers[this.curr]=number
            this.used[number]=false
        }
    }
}

obj=new PhoneDirectory(5)
console.log(obj.numbers)
console.log(obj.get())
console.log(obj.used)
console.log(obj.get())
console.log(obj.used)
console.log(obj.check(1))
obj.release(1)
console.log(obj.check(1))